packages/frontend/src/components/fields-editor/FieldsEditor.tsx

```diff
@@ -0,0 +1,215 @@
+  TextField,
+  Typography,
+} from "@mui/material";
+import { EditableBridgeConfig } from "../bridge/BridgeConfigEditor";
+import {
+  HomeAssistantMatcher,
+  HomeAssistantMatcherType,
+} from "@home-assistant-matter-hub/common";
+import Divider from "@mui/material/Divider";
+import AddCircleOutlineIcon from "@mui/icons-material/AddCircleOutline";
+import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
+import { useEffect, useState } from "react";
+
+interface FieldsEditorProps {
+  value: EditableBridgeConfig;
+  onChange: (event: { value: EditableBridgeConfig; isValid: boolean }) => void;
+}
+
+const FieldsEditor = ({ value, onChange }: FieldsEditorProps) => {
+  const [fieldValues, setFieldsValues] = useState<EditableBridgeConfig>(
+    () => value,
+  );
+
+  const handleOnChange = (
+    newIncludesValue: HomeAssistantMatcher[],
+    key: "include" | "exclude",
+  ) => {
+    const data = JSON.parse(JSON.stringify(value));
+    data.filter[key] = newIncludesValue;
+    setFieldsValues(data);
+  };
+
+  useEffect(() => {
+    if (JSON.stringify(value) === JSON.stringify(fieldValues)) return;
+
+    const includesFinted = fieldValues.filter.include.find(
+      (e) => !e.type || !e.value,
+    );
+    const excludeFinted = fieldValues.filter.exclude.find(
+      (e) => !e.type || !e.value,
+    );
+    if (includesFinted || excludeFinted) {
+      onChange({ isValid: false, value: fieldValues });
+      return;
+    }
+    onChange({ isValid: true, value: fieldValues });
+  }, [fieldValues, value, onChange]);
+
+  return (
+    <>
+      <EditionFilter
+        title="Includes"
+        values={fieldValues.filter.include}
+        onChange={(newIncludesValue) =>
+          handleOnChange(newIncludesValue, "include")
+        }
+      />
+      <EditionFilter
+        title="Excludes"
+        values={fieldValues.filter.exclude}
+        onChange={(newExcludesValue) =>
+          handleOnChange(newExcludesValue, "exclude")
+        }
+      />
+    </>
+  );
+};
+
+interface EditionFilterProps {
+  title: string;
+  values: HomeAssistantMatcher[];
+  onChange: (value: HomeAssistantMatcher[]) => void;
+}
+
+const EditionFilter = ({ title, values, onChange }: EditionFilterProps) => {
+  const handdleChange = (
+    newMatterValue: HomeAssistantMatcher,
+    index: number,
+  ) => {
+    const newValues = [...values];
+    newValues[index] = newMatterValue;
+    onChange(newValues);
+  };
+
+  const handdleOnAddField = () => {
+    const newValues = [...values];
+    newValues.push({
+      type: HomeAssistantMatcherType.Pattern,
+      value: "",
+    });
+    onChange(newValues);
+  };
+
+  const onDeleteElement = (index: number) => {
+    const newValues = [...values];
+    onChange(newValues.filter((_, i) => i !== index));
+  };
+
+  return (
+    <>
+      <Card
+        style={{
+          marginTop: "1rem",
+        }}
+      >
+        <CardContent>
+          <Grid2 container rowSpacing={2}>
+            <Grid2 size={12}>
```

Comment:
what's the point of using `Grid` when all children have full size? If its only about `rowSpacing`, use `Stack` instead.